BlueDBOnDisk/src/main/java/org/bluedb/disk/file/ReadWriteFileManager.java

```diff
@@ -86,24 +87,31 @@ protected void writeMetadata(BlueFileMetadata metadata, DataOutputStream dos, Pa
 		}
 	}
 
-	public void makeUnencryptedCopy(Path srcPath, Path destPath) throws BlueDbException, IOException {
+	public void makeCopy(Path srcPath, Path destPath) throws BlueDbException, IOException {
 		// Create file at destPath on the system if needed
 		destPath.toFile().getParentFile().mkdirs();
 		destPath.toFile().createNewFile();
-		
+
 		try (
 				DataInputStream dis = FileUtils.openDataInputStream(srcPath.toFile());
 				DataOutputStream dos = FileUtils.openDataOutputStream(destPath.toFile())
 		) {
-			// Write metadata with ENCRYPTION_VERSION_KEY removed
+			// Write metadata
 			BlueFileMetadata srcMetadata = readMetadata(dis);
-			BlueFileMetadata destMetadata = srcMetadata != null ? srcMetadata : new BlueFileMetadata();
-			destMetadata.remove(BlueFileMetadataKey.ENCRYPTION_VERSION_KEY);
+			BlueFileMetadata destMetadata = new BlueFileMetadata();
+			if (encryptionService.isEncryptionEnabled()) {
+				destMetadata.put(BlueFileMetadataKey.ENCRYPTION_VERSION_KEY, encryptionService.getCurrentEncryptionVersionKey());
+			}
 			writeMetadata(destMetadata, dos, destPath);
-
-			// Write unencrypted bytes
-			byte[] unencryptedBytes = encryptionService.decryptOrReturn(srcMetadata, FileUtils.readAllBytes(dis));
-			dos.write(unencryptedBytes);
+			
+			// Write bytes
+			byte[] rawBytes = FileUtils.readAllBytes(dis);
+			if (EncryptionUtils.shouldWriterSkipEncryptionForUnchangedDataUsingRawBytes(srcMetadata, destMetadata)) {
+				dos.write(rawBytes);
+			} else {
+				byte[] unencryptedBytes = encryptionService.decryptOrReturn(srcMetadata, rawBytes);
+				dos.write(unencryptedBytes);

```

Comment:
Shouldn't this be re-encrypting the bytes if the current encryption version is valid. The destination meta data definitely says that it uses the current encryption version key which could cause problems when reading the unencrypted bytes here.

Reply:
You are completely correct here, great catch! I missed updating this branch of the logic, updating that now.